### PR TITLE
101 add qr code scanner in android studio

### PR DESCRIPTION
**Pull request by FaizaAmjad**
⛓ **Description:**

- Helps #101

Add QR scan icon in the app to scan objects and display results in an alert dialog box connected with google search engine.

🛠 **Type of change:**
- [x] Scan objects by using camera.
- [x] Code is configured in Kotlin.
- [x] Icon in game screen to scan objects on click.
- [X] Show alerts box in the app in order to get permission from the user to open google search results.

### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Code change tested manually if not covered by automated tests
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change

### CODE REVIEW

File: GreenGarbageApp/app/src/main/res/layout/fragment_game.xml [hunk 1]
```diff
@@ -53,6 +53,12 @@
             android:background="@color/white"
             android:orientation="horizontal">
```
Comment (Toerngren) [inline GreenGarbageApp/app/src/main/res/layout/fragment_game.xml:55]: This space is currently full, after Umair added the trash count here. We need to move this to somewhere else.

File: GreenGarbageApp/app/src/main/java/com/example/greengarbageapp/activities/MainActivity.kt [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
+import android.content.DialogInterface
+import android.content.Intent
 import android.media.AudioManager
 import android.media.MediaPlayer
 import android.os.Bundle
 import android.view.View
-import android.widget.Button
+import android.widget.ImageButton
 import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
```
Comment (Toerngren) [inline GreenGarbageApp/app/src/main/java/com/example/greengarbageapp/activities/MainActivity.kt:12]: Could this be better implemented with a DialogBox instead?

File: GreenGarbageApp/app/src/main/java/com/example/greengarbageapp/activities/MainActivity.kt [hunk 3]
```diff
@@ -41,6 +46,32 @@ class MainActivity : AppCompatActivity(){
                 drawerLayout.setDrawerLockMode(DrawerLayout.LOCK_MODE_LOCKED_CLOSED)
             }
         }
+        // QR-code scanner
```
Comment (Toerngren) [inline GreenGarbageApp/app/src/main/java/com/example/greengarbageapp/activities/MainActivity.kt:49]: Why is this in MainActivity?

**Review by Toerngren — CHANGES_REQUESTED** [review]
Navigation nor app structure dosen't work. Sent images in Discord. DM me if you need more input.

File: GreenGarbageApp/app/src/main/res/layout/fragment_game.xml (hunk 1 above)
Comment (FaizaAmjad) [reply GreenGarbageApp/app/src/main/res/layout/fragment_game.xml:55]: I'll add a new linear layout.

File: GreenGarbageApp/app/src/main/java/com/example/greengarbageapp/activities/MainActivity.kt (hunk 2 above)
Comment (FaizaAmjad) [reply GreenGarbageApp/app/src/main/java/com/example/greengarbageapp/activities/MainActivity.kt:12]: The library uses the the Alert dialog box that's why I used the same.

File: GreenGarbageApp/app/src/main/java/com/example/greengarbageapp/activities/MainActivity.kt (hunk 3 above)
Comment (FaizaAmjad) [reply GreenGarbageApp/app/src/main/java/com/example/greengarbageapp/activities/MainActivity.kt:49]: I have looked into it and I realized it is skipping our intro page and going directly to game page. I am moving this to other class as you suggested.

**Comment by FaizaAmjad** [comment]
Will make some changes as it is giving some abnormalities.

**Pull request closed without merging**